use-package.el

```diff
@@ -463,13 +463,11 @@ manually updated package."
 
 (defun use-package-handler/:ensure (name keyword ensure rest state)
   (let ((body (use-package-process-keywords name rest state)))
-    ;; This happens at macro expansion time, not when the expanded code is
-    ;; compiled or evaluated.
-    (let ((package-name (or (and (eq ensure t) (use-package-as-symbol name)) ensure)))
-      (when package-name
-        (require 'package)
-        (use-package-ensure-elpa package-name)))
-    body))
+    `((let ((package-name (or (and (eq ,ensure t) (use-package-as-symbol ',name)) ,ensure)))
```

Comment:
@npostavs 

>Also, I think it should be ',(use-package-as-symbol name).

Hmm, I think it shouldn't.
If you do ` ',(use-package-as-symbol name)`, then it will be executed on each macro expansion. `use-package-as-symbol` interns symbols, which means that if you type `(use-package test123|` and do little pause between each letter, then it will call `use-package-as-symbol` many times and will intern symbols `t`, `te`, `tes`, `test` and so on.

Correct me if I'm wrong.

>You should probably use make-symbol for the package-name binding.

Not sure about this one as well.
`name` is a symbol, not a string. 
`make-symbol` requires a string.

How do you suggest to use `make-symbol` here?
Convert `name` (already interned by `use-package-as-symbol`) to a string and then `make-symbol` out of this string?